src/iterative_eval_abstracts/main.py

```diff
@@ -0,0 +1,170 @@
+import logging
+from pathlib import Path
+
+import pandas as pd
+import weaviate
+from fondant.pipeline import ComponentOp, Pipeline
+from fondant.pipeline.compiler import DockerCompiler
+from fondant.pipeline.runner import DockerRunner
+from iterative_eval_abstracts.read_output_data import read_latest_data
+
+logger = logging.getLogger(__name__)
+
+
+class IterativeEvaluation:
```

Comment:
I would prefer not to put this into a class, but update the `pipeline.py` and `pipeline_eval.py` class.

They should both look like this:

```python
def create_pipeline(**kwargs):
    # Create the pipeline

if __name__ == "__main__":
    pipeline = create_pipeline(**kwargs)
```

The in the notebook, I would do:
```python
import pipeline
import pipeline_eval

indexing_pipeline = pipeline.create_pipeline(**kwags)
eval_pipeline = pipeline_eval.create_pipeline(**kwargs)

runner = DockerRunner()  # Compiler is no longer needed
runner.run(indexing_pipeline)
runner.run(eval_pipeline)
```

Reply:
Oh I see, that's much better, it'll enable the user to modify the pipeline and not the arguments only. Will work on it.